File: java/src/bfs/MinDistLabyrinthTest.java
package bfs;

import org.junit.Assert;
import org.junit.Test;

import java.util.Arrays;

public class MinDistLabyrinthTest {
	@Test
	public void testRandomCases() {
		for (int i = 0; i < 100_000; i++) {
			var td = new randomLabyrinthTestData();

			var expected = td.minDist;
			var actual = MinDistLabyrinth.minDistLabyrinth(td.labyrinth);

			if (expected != actual) {
				Assert.fail(String.format(
					"minDistLabyrinth(%s):\nwant %s\n got %s",
					Arrays.deepToString(td.labyrinth), expected, actual
				));
			}
		}
	}
}

File: python/bfs/min_dist_labyrinth_test.py
from min_dist_labyrinth import min_dist_labyrinth
from random_labyrinth_test_data import RandomLabyrinthTestData


def test_min_dist():
	for _ in range(10_000):
		td = RandomLabyrinthTestData()

		want = td.min_dist
		got = min_dist_labyrinth(td.labyrinth)

		if got != want:
			assert False, f'minDistLabyrinth({td.labyrinth}):\nwant {want}\n got {got}'
